flight/extract_gps.py

```diff
@@ -0,0 +1,94 @@
+"""Contains the extract_gps() function for extracting data out of
+the provided waypoint data JSON file for the SUAS competition."""
+from collections import namedtuple
+from typing import Any, List, NamedTuple
+import json
+from typing_extensions import TypedDict
+
+GPSData = TypedDict(
+    "GPSData",
+    {
+        "waypoints": List[NamedTuple],
+        "boundary_points": List[NamedTuple],
+        "altitude_limits": List[int],
+    },
+)
+
+
+def extract_gps(path: str) -> GPSData:
+    """Returns the waypoints, boundary points, and altitude limits from a waypoint data file.
+
+    Args:
+        path (str): File path to the waypoint data JSON file.
+
+    Returns:
+        GPSData: TypedDict[
+            list(NamedTuple(int, int, int)), list(NamedTuple(int, int)), list(int, int)
+        ]
+            The data in the waypoint data file
+            waypoints : list(NamedTuple(int, int, int))
+                Waypoint : NamedTuple(int, int, int)
+                    latitude : int
+                        The latitude of the waypoint.
+                    longitude : int
+                        The longitude of the waypoint.
+                    altitude : int
+                        The altitude of the waypoint.
+            boundary_points : list(NamedTuple(int, int, int))
+                BoundaryPoint : NamedTuple(int, int, int)
+                    latitude : int
+                        The latitude of the boundary point.
+                    longitude : int
+                        The longitude of the boundary point.
+            altitude_limits : list(int, int)
+                altitude_min : int
+                    The minimum altitude that the drone must fly at all times.
+                altitude_max : int
+                    The maximum altitude that the drone must fly at all times.
+    """
+    # Initialize namedtuples to store latitude/longitude/altitude data for provided points
+    Waypoint = namedtuple("Waypoint", ["latitude", "longitude", "altitude"])
+    BoundaryPoint = namedtuple("BoundaryPoint", ["latitude", "longitude"])
```

Comment:
This is actually the legacy way of creating named tuples. The new "proper way" is to import NamedTuple from typing and construct a class that inherits it. See https://missourimrr.github.io/docs/python/type-annotations/full-guide.html#namedtuples